Add LiteSVM Tests for Checking Accounts (Anchor)

sol-warrior opened the pull request:
This PR modernizes the **`checking-accounts Anchor example`** by replacing deprecated Bankrun tests with a new LiteSVM test suite and aligning the project with current Solana testing practices as @Perelyn-sama mentioned in prev PR , bankrun is deprecated.

### Changes Included
- Added LiteSVM tests covering the **checking-accounts** workflow.
- Updated TypeScript config (module system, target, JSON resolution).
- Added **LiteSVM** dependencies and updated pnpm-lock.yaml.
- Added a LiteSVM test script in `Anchor.toml`.
- Removed **deprecated Bankrun** dependencies and old test file.

### Request
@jacobcreech When you have a moment, I’d appreciate your review on this update.
Any suggestions or adjustments are most welcome — I’m happy to refine anything as needed.

Thank you for your time and support 🙌🏼

sol-warrior committed:
feat: add LiteSVM tests for checking accounts functionality

sol-warrior committed:
chore: update TypeScript configuration for checking accounts anchor

- Change module system to 'nodenext'
- Update target to 'es2020'
- Enable JSON module resolution

sol-warrior committed:
chore: update dependencies for checking accounts anchor

- Upgrade @solana/web3.js to version 1.98.4
- Add litesvm as a new dependency with version 0.4.0
- Update pnpm-lock.yaml to reflect changes in dependency versions and resolutions

sol-warrior committed:
chore: add litesvm test script to Anchor.toml

- Introduce a new test script for LiteSVM in the Anchor.toml configuration file.

sol-warrior committed:
chore: remove unused dependencies and test file for bankrun

- Remove anchor-bankrun and solana-bankrun dependencies from package.json and pnpm-lock.yaml.
- Delete the bankrun.test.ts file as it is no longer needed, It's deprecated.

Perelyn-sama commented on basics/checking-accounts/anchor/Anchor.toml:
```diff
@@ -12,3 +12,4 @@ wallet = "~/.config/solana/id.json"
 
 [scripts]
 test = "pnpm ts-mocha -p ./tsconfig.json -t 1000000 tests/**/*.ts"
+litesvm-test = "node --test tests/litesvm.test.ts"
```
if `litesvm.test.ts` is the only test, what's the point of having two scripts that run it. 

just the test script should be fine.

Perelyn-sama commented:
hey @sol-warrior could please finish this up? remove bankrun, just use litesvm

sol-warrior committed:
chore: fix ts error and rename the test description

- Change the import path for CheckingAccountProgram to include the .ts extension.
- Rename the test suite description to "Anchor: Checking accounts" for clarity.

sol-warrior committed:
chore: remove litesvm test script and update litesvm test to use chai assertion library

- Replace Node's strict assert with Chai's assert for improved testing capabilities.
- Remove unused imports in litesvm.test.ts for cleaner code.
- Ensure consistency in Anchor.toml by maintaining the test script format.
